Sources/Subs-Db-postgresql.php

```diff
@@ -648,9 +648,15 @@ function smf_db_insert($method = 'replace', $table, $columns, $data, $keys, $ret
 	$table = str_replace('{db_prefix}', $db_prefix, $table);
 
 	// Sanity check for replace is key part of the columns array
-	if ($method == 'replace' && count(array_intersect_key($columns, array_flip($keys))) !== count($keys))
-		smf_db_error_backtrace('Primary Key field missing in insert call',
-			'Change the method of db insert to insert or add the pk field to the columns array', E_USER_ERROR, __FILE__, __LINE__);
+	if ($method == 'replace')
+	{
+		if (empty($keys))
+			smf_db_error_backtrace('When using the replace mode, the key column is a required entry.',
+				'Change the method of db insert to insert or add the pk field to the key array', E_USER_ERROR, __FILE__, __LINE__);
```

Comment:
Why are texts of error messages not to be translated?

Reply:
So the same reason like they are in other places in db layer not able to be translated.
- DB layer is called very very early (no information about the user exists)
- To post not english version of this error in the support board doesn't help
- People how are not able to read the english text are not able to solve this kind of issue